### PR TITLE
Bugfix/re throw error in http intercepter

### PR DESCRIPTION
**Pull request by danoswaltCL**
as discussed in some mtg, we should allow this error keep moving through so the error can be caught in angular components

### CODE REVIEW

File: frontend/projects/upgrade/src/app/core/http-interceptors/http-error.interceptor.ts [hunk 1]
```diff
@@ -32,7 +32,9 @@ export class HttpErrorInterceptor implements HttpInterceptor {
           this.authService.authLogout();
         }
         this.openPopup(err);
-        return EMPTY; // returning EMPTY instead of throwError as Error is handled using snacker here itself
+
+        // re-throw to allow the error to be caught by the calling code
+        return throwError(() => new Error(JSON.stringify(err)));
```
Comment (VivekFitkariwala) [inline frontend/projects/upgrade/src/app/core/http-interceptors/http-error.interceptor.ts:37]: Why not just `throw error`?

**Review by VivekFitkariwala — CHANGES_REQUESTED** [review]

File: frontend/projects/upgrade/src/app/core/http-interceptors/http-error.interceptor.ts (hunk 1 above)
Comment (danoswaltCL) [reply frontend/projects/upgrade/src/app/core/http-interceptors/http-error.interceptor.ts:37]: I'm sorry, what is the code suggestion to try?

Comment (danoswaltCL) [reply frontend/projects/upgrade/src/app/core/http-interceptors/http-error.interceptor.ts:37]: ok I see just `throw err` does work, let me fix the tests.

Comment (danoswaltCL) [reply frontend/projects/upgrade/src/app/core/http-interceptors/http-error.interceptor.ts]: fixed @VivekFitkariwala

**Comment by danoswaltCL** [comment]
@VivekFitkariwala is this looking right to you now? I'll need this fixed for the demo.

**Review by VivekFitkariwala — APPROVED** [review]

**Comment by VivekFitkariwala** [comment]
@danoswaltCL yup approved

**Pull request merged**